wgpu/src/api/compute_pipeline.rs

```diff
@@ -20,6 +20,13 @@ super::impl_partialeq_eq_hash!(ComputePipeline);
 
 impl ComputePipeline {
     /// Get an object representing the bind group layout at a given index.
+    ///
+    /// If this pipeline was created with a [default layout][ComputePipelineDescriptor::layout],
+    /// then bind groups created with the returned `BindGroupLayout` can only be used with this
+    /// pipeline.
+    ///
+    /// # Panics
+    /// - There is no bind group layout at the given `index`.
```

Comment:
urgh this _actually_ panics. Wasn't even aware.
For future reference: This can be a `Result`. WebGpu backend has to handle the js exception for this https://developer.mozilla.org/en-US/docs/Web/API/GPUComputePipeline/getBindGroupLayout#validation

Reply:
> this *actually* panics

By this, do you mean that there's a bug here? I.e. `wgpu` should be catching a validation error and doing the proper thing with it, but it's not?

Or do you just mean to say "ah yes, I suppose this *does* panic if the user doesn't set any error scopes."

Reply:
Looked closer again to be sure: Today it panics on wgpu-core which is a simply a bug. Turns out this also got filed already (didn't check before).. twice in fact. 
* https://github.com/gfx-rs/wgpu/issues/4167
* https://github.com/gfx-rs/wgpu/issues/6095

...closing one of them as dupl

Considering that, I think the doc here should say that this is a validation error and then add a note that there is a known bug that this panics

Reply:
Just FYI, I actually didn't go and research for myself whether it panics. I just made an assumption and hoped that my code reviewer would correct me.

Looking at the implementation, it looks like it does the proper thing on WebGPU, but not on wgpu_core. Given that, I don't think we should put anything in the public docs until we fix wgpu_core to raise this as a validation error.

Reply:
Oops, I responded before I refreshed and saw your reply. The known bug option sounds fine too.

Reply:
Okay actually I changed my mind, I'm going to omit any documentation about panics or errors until webgpu and wgpu-core have the same behavior.

Reply:
fine by me! Didn't want to bog down this otherwise fine doc addition :)
Bit of a wild west situation this particular thing at this point T_T